### ISSUE TITLE
Convenience methods stampit.state(), stampit.methods(), stampit.enclose()

### ISSUE THREAD

**Issue by ericelliott** [ericelliott/stampit#41]
Normally you have to type `stampit().state()` etc.. to start a new stamp. Sometimes it's easy to forget that you have to invoke `stampit` as a function before you have access to those methods.

It would be convenient if there were convenience methods on `stampit` that would create a new stamp automatically.

**Comment by koresar** [comment]
What do you mean by "convenience methods"? A code sample would be nice.

**Comment by ericelliott** [comment]
e.g., instead of:

```js
var stamp = stampit().state(state);
```

You should be able to do:

```js
var stamp = stampit.state(state);
```

**Comment by koresar** [comment]
I think this is not worth the change. Two characters `()` is not something what bothers me. :) Let me push a PR. I'll explain my concerns there.

**Comment by ericelliott** [comment]
I've interacted with a lot of Stampit users. Trust me when I say it sometimes bothers people. This was actually a feature request from [Respond.ly](https://respond.ly/), who recently started using Stampit.

**Comment by koresar** [comment]
Then let's add them. :)

![Add all the features!](http://treasure.diylol.com/uploads/post/image/513436/resized_all-the-things-meme-generator-add-all-the-features-137487.jpg)

**Comment by ericelliott** [comment]
:+1:

**Comment by JosephClay** [comment]
My only concern is that `stampit.state(state)` implies that state is being configured on `stampit` and not the returned stamp; but the docs are pretty clear on stampit returning factory functions. Convenience over explicitness?

**Comment by ericelliott** [comment]
@JosephClay Yeah, since we actually know that this causes bugs and inconvenience for our users without the convenience methods, I think we already have our answer on this. =)

**Comment by ericelliott** [comment]
Hey, is this feature in the Stampit 2.0 branch, yet?

Maybe it should be. =)

**Comment by koresar** [comment]
It can go to both 1.x and 2.x. What do you reckon?

**Comment by koresar** [comment]
As I mentioned, @ericelliott, I would implement every other feature in the world as soon as we merge the 'immutability' PR to the `v2_0` branch. Otherwise, it will be just too many merge conflicts. Would you mind approving #48?

**Comment by koresar** [comment]
Implemented with #84.
